### ISSUE TITLE
feat: support non-top-level partials

### ISSUE THREAD

**Issue by alestiago** [felangel/mason#378]
**Description**

Allow partials to be placed elsewhere than under `__brick__` directory.

**Requirements**

- [ ] Allow partials not under `__brick__`

**Comment by talamaska** [comment]
Now, not to be picky, but the "Allow partials not under ```__brick__```". To me sounds like somewhere outside of the ```__brick__``` folder outside. 
I would say that I want to be able to have partials working somewhere nested under ```__brick__```, like 
```__brick__``` > feature_name > bloc > here. Is this the same that this issue is requesting?

**Comment by felangel** [comment]
> Now, not to be picky, but the "Allow partials not under `__brick__`". To me sounds like somewhere outside of the `__brick__` folder. I would say that I want to be able to have partials working somewhere nested under `__brick__`, like `__brick__` > feature_name > bloc > here. Is this the same that this issue is requesting?

Yes, I believe so. I believe this issue is requesting that partials be placed anywhere within the `__brick__` directory (even in nested directories)

**Comment by talamaska** [comment]
But I think they still need to be on the same level of the template that is using them.
